pastas/stats/metrics.py

```diff
@@ -353,6 +354,57 @@ def nse(
     return 1 - (w * err.to_numpy() ** 2).sum() / (w * (obs.to_numpy() - mu) ** 2).sum()
 
 
+def nnse(
+    obs: Series,
+    sim: Optional[Series] = None,
+    res: Optional[Series] = None,
+    missing: str = "drop",
+    weighted: bool = False,
+    max_gap: int = 30,
+) -> float:
+    """Compute the (weighted) Normalized Nash-Sutcliffe Efficiency (NNSE).
+
+    Parameters
+    ----------
+    obs: pandas.Series
+        Series with the observed values.
+    sim: pandas.Series, optional
+        The Series with the simulated values.
+    res: pandas.Series, optional
+        The Series with the residual values. If time series for the residuals are
+        provided, the sim and obs arguments are ignored. Note that the residuals
+        must be computed as `obs - sim` here.
+    missing: str, optional
+        string with the rule to deal with missing values. Only "drop" is supported now.
+    weighted: bool, optional
+        If weighted is True, the variances are computed using the time step between
+        observations as weights. Default is False.
+    max_gap: int, optional
+        maximum allowed gap period in days to use for the computation of the weights.
+        All time steps larger than max_gap are replace with the max_gap value.
+        Default value is 30 days.
+
+    Notes
+    -----
+    NNSE computed according to :cite:t:`nash_normalized_2006`
+
+    .. math:: \\text{NNSE} = 1 / (2 - NSE)
+
```

Comment:
But can you add a sentence here when the NNSE should be preferred over the NSE? Just to guide users a bit on its usage.